src/main/webapp/app/shared/modal/ModifyTherapyModal.tsx

```diff
@@ -105,8 +110,28 @@ const ModifyTherapyModalContent = observer(
         title={treatmentUuid.startsWith('new_treatment_for') ? 'Add Therapy(s)' : 'Modify Therapy(s)'}
         show={true}
         onConfirm={() => {
-          const name = modifyTherapyModalStore.selectedTreatments.map(therapy => therapy.map(drug => drug.uuid).join(' + ')).join(', ');
-          onConfirm(name);
+          const newDrugs: IDrug[] = modifyTherapyModalStore.selectedTreatments.reduce((accumulator: IDrug[], currentTreatment) => {
+            const drugs = currentTreatment
+              .filter(therapy => therapy.ncit)
+              .map<IDrug>(therapy => ({
+                uuid: generateUuid(),
+                name: therapy.ncit.preferredName,
+                nciThesaurus: parseNcitUniqId(therapy.value),
+              }));
+            return accumulator.concat(drugs);
+          }, []);
```

Comment:
This newDrugs will have dups if I add the same drug in different treatments. Only one drug creation call will success as expected. But other ones will fail and the warning message will pop up.